src/Application/Actions/RegularGivingMandate/Create.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace MatchBot\Application\Actions\RegularGivingMandate;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Laminas\Diactoros\Response\JsonResponse;
+use MatchBot\Application\Actions\Action;
+use MatchBot\Application\AssertionFailedException;
+use MatchBot\Application\Auth\PersonWithPasswordAuthMiddleware;
+use MatchBot\Application\Environment;
+use MatchBot\Application\HttpModels\MandateCreate;
+use MatchBot\Domain\CampaignRepository;
+use MatchBot\Domain\PersonId;
+use MatchBot\Domain\RegularGivingMandate;
+use MatchBot\Domain\Salesforce18Id;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Log\LoggerInterface;
+use Slim\Exception\HttpBadRequestException;
+use Slim\Exception\HttpNotFoundException;
+use Symfony\Component\Serializer\Exception\UnexpectedValueException;
+use Symfony\Component\Serializer\SerializerInterface;
+
+class Create extends Action
+{
+    public function __construct(
+        private Environment $environment,
+        LoggerInterface $logger,
+        private CampaignRepository $campaignRepository,
+        private SerializerInterface $serializer,
+        private EntityManagerInterface $em,
+    ) {
+        parent::__construct($logger);
+    }
+
+    protected function action(Request $request, Response $response, array $args): Response
+    {
+        if (! $this->environment->isFeatureEnabledRegularGiving()) {
+            throw new HttpNotFoundException($request);
+        }
+
+        $donorIdString = $request->getAttribute(PersonWithPasswordAuthMiddleware::PERSON_ID_ATTRIBUTE_NAME);
+        \assert(is_string($donorIdString));
+        $donorID = PersonId::of($donorIdString);
+
+        $body = (string) $request->getBody();
+
+        try {
+            $mandateData = $this->serializer->deserialize($body, MandateCreate::class, 'json');
+            \assert($mandateData instanceof MandateCreate);
+        } catch (\TypeError | UnexpectedValueException | AssertionFailedException $exception) {
+            /** similar catch with commentary in @see \MatchBot\Application\Actions\Donations\Create */
+            $this->logger->info("Mandate Create non-serialisable payload was: $body");
+
+            $message = 'Donation Create data deserialise error';
+            $exceptionType = get_class($exception);
+
+            return $this->validationError(
+                $response,
+                "$message: $exceptionType - {$exception->getMessage()}",
+                $message,
+                empty($body), // Suspected bot / junk traffic sometimes sends blank payload.
```

Comment:
I think this is hopefully less likely with this new, more niche endpoint. Should we start with error level logs until we know otherwise?

Reply:
yes I think that's fine. I think I will also change the middleware now to be PersonWithPassword which will probably make bots a lot less likely to access it.

Reply:
Done in 822e9d91